NoSuchElementException when building a project with no source files

## Describe the bug
I am trying to plugin libraries via CDI into a service to create a new service. So in my example my original service is `service1`, which I'd like to add `plugin2` to create `service2`. So I made a new project with no source code and only dependencies in `service1` and `plugin2`. I was hoping the REST resources from `service1` would be picked up off the classpath along with `plugin2` through CDI. But when I `quarkusBuild`, I get a `NoSuchElementException`.

### Expected behavior
A successful build.

### Actual behavior
```
Caused by: java.util.NoSuchElementException
        at io.quarkus.deployment.builditem.ArchiveRootBuildItem.<init>(ArchiveRootBuildItem.java:97)
        at io.quarkus.deployment.builditem.ArchiveRootBuildItem.<init>(ArchiveRootBuildItem.java:17)
        at io.quarkus.deployment.builditem.ArchiveRootBuildItem$Builder.build(ArchiveRootBuildItem.java:50)
        at io.quarkus.deployment.QuarkusAugmentor.run(QuarkusAugmentor.java:141)
        at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:394)
        ... 125 more
```


## To Reproduce

`service2` in https://github.com/mweber03/quarkus-test

Steps to reproduce the behavior:
1. Uncomment `service2` from `settings.gradle`
2. `./gradlew service2:build`

## Environment (please complete the following information):

### Output of `uname -a` or `ver`
Ubuntu 20.04

### Output of `java -version`
OpenJDK 11.0.11

### Quarkus version or git rev
Quarkus 1.13.7

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)
Gradle 6.8.3

--- a/core/deployment/src/main/java/io/quarkus/deployment/builditem/ArchiveRootBuildItem.java
+++ b/core/deployment/src/main/java/io/quarkus/deployment/builditem/ArchiveRootBuildItem.java
@@ -81,20 +81,25 @@ public final class ArchiveRootBuildItem extends SimpleBuildItem {
 
     private ArchiveRootBuildItem(Builder builder, QuarkusBuildCloseablesBuildItem buildCloseables) throws IOException {
         this.excludedFromIndexing = builder.excludedFromIndexing;
-        final PathsCollection.Builder rootDirs = PathsCollection.builder();
-        final PathsCollection.Builder paths = PathsCollection.builder();
-        for (Path root : builder.archiveRoots) {
-            paths.add(root);
-            if (Files.isDirectory(root)) {
-                rootDirs.add(root);
-            } else {
-                final FileSystem fs = buildCloseables.add(FileSystems.newFileSystem(root, null));
-                fs.getRootDirectories().forEach(rootDirs::add);
+        if (!builder.archiveRoots.isEmpty()) {
+            final PathsCollection.Builder rootDirs = PathsCollection.builder();
+            final PathsCollection.Builder paths = PathsCollection.builder();
+            for (Path root : builder.archiveRoots) {
+                paths.add(root);
+                if (Files.isDirectory(root)) {
+                    rootDirs.add(root);
+                } else {
+                    final FileSystem fs = buildCloseables.add(FileSystems.newFileSystem(root, null));
+                    fs.getRootDirectories().forEach(rootDirs::add);
+                }
             }
+            this.rootDirs = rootDirs.build();
+            this.paths = paths.build();
+            this.archiveRoot = this.rootDirs.iterator().next();
+        } else {
+            this.paths = this.rootDirs = PathsCollection.of();
+            this.archiveRoot = null;
         }
-        this.rootDirs = rootDirs.build();
-        this.paths = paths.build();
-        this.archiveRoot = this.rootDirs.iterator().next();
     }
 
     /**

